File: Level1/소수 만들기/소수 만들기.py
import Math
def solution(nums):
    answer = 0
    
    for i in range(len(nums)):
        for j in range(i+1,len(nums)):
            for k in range(j+1,len(nums)):
                print(nums[i]+nums[j]+nums[k])

    return answer

def prime(number):
    if(number == 0 or number == 1):
        return false
    
    for i in range(2,Math.sqrt(number)):
        if(number%i==0)

File: Level1/소수 만들기/소수 만들기.java
import java.util.*;
class Solution {
    ArrayList<Integer> list = new ArrayList<Integer>();
    public int solution(int[] nums) {
        int answer = 0;
        boolean [] visited = new boolean[nums.length];
        combination(nums,visited, 0, 3);
        for(int i=0; i<list.size(); i++){
            if(prime(list.get(i))){
                answer++;
            }
        }
        return answer;
    }
    void combination(int[] arr, boolean[] visited, int start, int r) {
        int sum=0;
        if (r==0){
            for(int i=0; i<arr.length; i++){
                if(visited[i]){
                    sum+=arr[i];
                }
            }
            list.add(sum);
            return;
        }

        for(int i=start; i<arr.length; i++){
            visited[i] = true;
            combination(arr, visited, i+1, r-1);
            visited[i] = false;
        }
        return;
    }
    
    boolean prime(int n){ 
        if(n==0 || n==1){
            return false;
        }
        for(int i=2; i<=(int)Math.sqrt(n); i++){
            if(n%i==0){
                return false;
            }
        }
        return true;
    }
    
}
